source/src/localisation/cs_NearestCrownstoneTracker.cpp

```diff
@@ -139,12 +128,19 @@ NearestWitnessReport NearestCrownstoneTracker::createReport(TrackableEvent* trac
 }
 
 // REVIEW: Return by ref?
+// @Bart: RVO (return value optimization) is guaranteed to take place
+// as the constructor call is part of the return statement.
+// (This means the return value is constructed on the stack frame below the current one.)
```

Comment:
Does it hurt to return by ref in that case? If the code changes, the optimization might be gone.

Reply:
Returning refs to locally constructed objects leads to undefined behaviour so you'ld have to pass in a reference to the location where the object needs to be constructed like C-style return-arguments. RVO does this automatically and can even skip multiple stack frames with 0 maintenance. Note that createReport(TrackableEvent*) uses the exact same paradigm. (Quick explanation: https://www.youtube.com/watch?v=IZbL-RGr_mk )